src/pudl/extract/csv.py

```diff
@@ -1,72 +1,121 @@
+        # str_part = str(list(partition.values())[0])
+        # col_map = self.METADATA._column_map[page]
+        # page_cols = col_map.loc[
+        #     (col_map[str_part].notnull()) & (col_map[str_part] != -1),
+        #     [str_part],
+        # ].index
+        # expected_cols = page_cols.union(self.cols_added)
+        # if set(newdata.columns) != set(expected_cols):
+        #     # TODO (bendnorman): Enforce canonical fields for all raw fields?
+        #     extra_raw_cols = set(newdata.columns).difference(expected_cols)
+        #     missing_raw_cols = set(expected_cols).difference(newdata.columns)
+        #     if extra_raw_cols:
+        #         logger.warning(
+        #             f"{page}/{str_part}:Extra columns found in extracted table:"
+        #             f"\n{extra_raw_cols}"
+        #         )
+        #     if missing_raw_cols:
+        #         logger.warning(
+        #             f"{page}/{str_part}: Expected columns not found in extracted table:"
+        #             f"\n{missing_raw_cols}"
+        #         )
+        df = pd.concat(dfs, sort=True, ignore_index=True)
+
+        # TODO: We're not expecting missing cols based on existing data
+        #  but might want to have this as abstracted functionality/contract
+        # After all years are loaded, add empty columns that could appear
+        # in other years so that df matches the database schema
+        # missing_cols = list(
+        #     set(self._metadata.get_all_columns(page)).difference(df.columns)
+        # )
+        # df = pd.concat([df, pd.DataFrame(columns=missing_cols)], sort=True)
+
+        # TODO: I don't think we need this unless we want to consolidate with Excel/generic extractor
+        # raw_dfs[page] = self.process_final_page(df=df, page=page)
+        raw_dfs[page] = df
+        return raw_dfs
+
+    def load_csv_file(self, **partition) -> pd.DataFrame:
+        """Produce the dataframe object for the given partition.
 
         Args:
-            zipfile: zipfile object containing source files
-            table_file_map: map of table name to source file in zipfile archive
+            partition: partition to load. (ex: 2009)
+
+        Returns:
+            pd.DataFrame instance containing CSV data
         """
-        self._zipfile = zipfile
-        self._table_file_map = table_file_map
-
-    def get_table_names(self) -> list[str]:
-        """Returns list of tables that this extractor provides access to."""
-        return list(self._table_file_map)
-
-    def extract_one(self, table_name: str) -> pd.DataFrame:
-        """Read the data from the CSV source file and return as a dataframe."""
-        logger.info(f"Extracting {table_name} from CSV into pandas DataFrame.")
-        filename = self._table_file_map[table_name]
-        with self._zipfile.open(filename) as f:
+        filename = self.csv_filename(partition["year"])
+
+        with self.ds.get_zipfile_resource(self._dataset_name) as zf, zf.open(
```

Comment:
Unlike the Excel extractor and the earlier incarnation of the CSV extractor, the source contents aren't zipped into a folder but instead into a top-level CSV file. I'll check on whether that makes a difference (i.e., in Dagster testing).